backend/backendutil/bodystructure.go

```diff
@@ -1,14 +1,40 @@
 package backendutil
 
 import (
+	"bufio"
+	"bytes"
 	"io"
+	"io/ioutil"
 	"mime"
 	"strings"
 
 	"github.com/emersion/go-imap"
 	"github.com/emersion/go-message/textproto"
 )
 
+type countReader struct {
+	r           io.Reader
+	total       uint32
+	newlines    uint32
+	lastNewline bool
+}
+
+func (r *countReader) Read(b []byte) (int, error) {
+	n, err := r.r.Read(b)
+	r.total += uint32(n)
+	if n != 0 {
+		r.newlines += uint32(bytes.Count(b[:n], []byte{'\r'}))
```

Comment:
Any reason why we count `\r` instead of `\n` here?

Reply:
Woops, right. It is supposed to be `\n` but I forgot to change it back after checking something.